src/main/java/sky/command/DeadlineCommand.java

```diff
@@ -0,0 +1,107 @@
+package sky.command;
+import java.io.IOException;
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.regex.PatternSyntaxException;
+
+import sky.TaskList;
+import sky.exception.TextNoMeaningException;
+import sky.storage.History;
+import sky.storage.Storage;
+import sky.task.Deadline;
+import sky.task.Task;
+
+/**
+ * The DeadlineCommand class deals with adding a deadline task into taskList.
+ */
+public class DeadlineCommand extends Command {
+    private String fullCommand;
+
+    public DeadlineCommand(String fullCommand) {
+        this.fullCommand = fullCommand;
+    }
+
+    @Override
+    public String execute(TaskList taskList, Storage storage, History history)
+            throws TextNoMeaningException, IOException {
+        try {
+            String[] arrOfStrings = generateTaskDescriptionAndUserInputBy();
+            assert arrOfStrings.length > 0 : "arrOfStrings should not be empty.";
+            String taskDescription = arrOfStrings[0];
+            String taskByUserInput = arrOfStrings[1];
+            String taskBy = produceDateAndTimeForDeadline(taskByUserInput);
+            Task task = new Deadline(taskDescription, taskBy);
+            taskList.addTask(task);
+            // Add task into data file.
+            storage.append(task.toString());
+            history.addHistoryInTime(taskList);
+            return generateResponse(task, taskList);
+        } catch (IndexOutOfBoundsException e) {
+            throw new TextNoMeaningException("You have either not entered any text after typing deadline, "
+                    + "or you have positioned your slash wrongly.");
+        } catch (PatternSyntaxException e) {
+            throw new TextNoMeaningException("There is a problem with the regex expression written by the dev.");
+        }
+    }
+
+    @Override
+    public boolean isExit() {
+        return false;
+    }
+
+    private String produceDateAndTimeForDeadline(String s) throws TextNoMeaningException {
```

Comment:
Should there be a JavaDoc comment for this method?

Reply:
I agree. Although JavaDocs is not a necessity for private methods, it's a good thing to have them for all methods. Good point!